spec/rubocop/cop/style/double_pipe_equals_spec.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+describe RuboCop::Cop::Style::DoublePipeEquals do
+  let(:config) { RuboCop::Config.new }
+  subject(:cop) { described_class.new(config) }
+
+  context 'when using var = var ? var : something' do
+    it 'registers an offense' do
+      expect_offense(<<-RUBY.strip_indent)
+        foo = foo ? foo : 'default'
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use the double pipe equals operator `||=` instead.
+      RUBY
+    end
```

Comment:
Could you add a test for keyword `if`? This cop adds offense to keyword `if` also.
For example:

```ruby
expect_offense(<<-RUBY).strip_indent
x = if x
      x
    else
      10
    end
RUBY
```